tuta-sdk/rust/sdk/src/services/generator.rs

```diff
@@ -0,0 +1,77 @@
+#[macro_export]
+macro_rules! __service_input_type {
+	(()) => {
+		$crate::services::hidden::Nothing
+	};
+	($i:ident) => {
+		$i
+	};
+}
+
+#[macro_export]
+macro_rules! __service_input_value {
+	($data:tt, ()) => {
+		None
+	};
+	($data:tt, $i: tt) => {
+		Some($data)
+	};
+}
+
+#[macro_export]
+macro_rules! __service_handle_response {
+	($x:expr, $res:expr, ()) => {
+		return Ok(());
+	};
+	($x:expr, $res:expr, $i: tt) => {
+		return $x.handle_response::<Self::Output>($res).await;
+	};
+}
+
+#[macro_export]
+macro_rules! service_impl {
```

Comment:
these macros looks a bit cryptic to me. I do not understand what `base` is. There's also no documentation on them either. Would be nice to document them (at least the interface but maybe also the impl) for those of us who are not so versed in arcane arts like me :sweat_smile:  🔮

Reply:
It's also two macros with the same name? Should they have different names perhaps?

Reply:
There is now further explanitation on how this macro is supposed to be used. doc is marked as:
```
```ignored
```
Without `ignored` cargo test will fail,
Since this macro expansion depends on `::services::service_executor::hidden` module which is not public.